concordium-std/src/impls.rs

```diff
@@ -638,21 +691,18 @@ where
     }
 
     pub fn is_empty(&self) -> bool {
-        if let Some(_) = self.state_ll.borrow().iterator(&self.prefix).next() {
-            false
-        } else {
-            true
-        }
+        let iterator = self.state_ll.borrow().iterator(&self.prefix);
+        iterator.is_err() || iterator.unwrap().next().is_none()
     }
 
-    pub fn iter(&self) -> StateMapIter<K, V, S> {
-        let state_iter = self.state_ll.borrow().iterator(&self.prefix);
-        StateMapIter {
+    pub fn iter(&self) -> Result<StateMapIter<K, V, S>, ContractStateError> {
+        let state_iter = self.state_ll.borrow().iterator(&self.prefix)?;
```

Comment:
`StateMapIter` should keep a reference to the `StateMap` so that this method can't be called twice without dropping the `StateMapIter`. When that is implemented, we can unwrap this call to `iterator`.